presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -139,7 +142,18 @@ public void start(HiveSplitSource splitSource)
     {
         this.hiveSplitSource = splitSource;
         for (int i = 0; i < maxPartitionBatchSize; i++) {
-            ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            UserGroupInformation currentHdfsUser = getCurrentHdfsUser();
+            ResumableTasks.submit(executor, new HiveSplitLoaderTask(currentHdfsUser));
+        }
+    }
+
+    private UserGroupInformation getCurrentHdfsUser()
+    {
+        try {
+            return UserGroupInformation.getCurrentUser();
```

Comment:
This method has a lot of downstream effects.  For example, if there is no subject attached to the thread, it will perform a login in the static `UGI` code.  Instead I think we need a way to only use this method when we know there is a kerberos subject associated with the thread.  For example:
```
    public UserGroupInformation getCurrentUser()
    {
        AccessControlContext context = AccessController.getContext();
        Subject subject = Subject.getSubject(context);
        if (subject == null || subject.getPrincipals(KerberosPrincipal.class).isEmpty()) {
            return null;
        }
        return UserGroupInformation.getUGIFromSubject(subject);
    }
```
Of course the bigger issue is that older versions of Hadoop do not have this method, so we will most likely need to add a method to the Hadoop shims (presto-hive-apache) that make back ports that method.